helpers/store/compounds.js

```diff
@@ -6,14 +7,17 @@ function callActionWhen (predicate) {
       return action
     }
 
-    knitLogger.debug(() => `Predicate: ${JSON.stringify(predicate)} evaluated to false, no action called!`)
-    this.methods.cancel()
+    knitLogger.debug(() => `Predicate: ${predicate} evaluated to false, canceling action!`)
+
+    // Cancel action
+    throw new ActionCancel(this.that.actionName)
   }
 }
 
 function doSth (fn) {
```

Comment:
Wow, how do I miss naming this function as it is? :( Please change it to proper name, and add description what it should be used for. BTW Is it really needed?